test-data/unit/check-enum.test

```diff
@@ -894,13 +894,13 @@ else:
 
 [case testEnumReachabilityWithNone]
 # flags: --strict-optional
-from enum import Enum
+from enum import Flag
```

Comment:
Why do we need this change?

Reply:
Hmm... let me think. I think it's because `Flag` implements `__bool__`, so with base `Enum` the following change would take place:
```diff
 if x:
     reveal_type(x)  # N: Revealed type is "__main__.Foo"
 else:
-     reveal_type(x)  # N: Revealed type is "Union[__main__.Foo, None]"
+     reveal_type(x)  # N: Revealed type is "None"
```
This is probably a vestige from my previous approach where lack of bool overload was taken as a signal that the class cannot be false-y. In this implementation I only emit a notice, but it doesn't affect always-true/false checks.

Long story short, I'll revert this change.